libsmtutil/CHCSmtLib2Interface.h

```diff
@@ -66,7 +66,7 @@ class CHCSmtLib2Interface: public CHCSolverInterface
 	std::string m_accumulatedOutput;
 	std::set<std::string> m_variables;
 
-	std::map<util::h256, std::string> const& m_queryResponses;
+	std::map<util::h256, std::string> m_queryResponses;
```

Comment:
Now it's a full copy of potentially large responses though :p

Reply:
The constructor got a copy (i.e. the responses by value) anyways and it's moved from there.
I can have a closer look at the call sites as well to see, if stuff is or can be moved from there, or if it actually has a long-enough life-time outside this class anyways, then we can pass it in to the constructor by reference and store it as reference here...

Reply:
Ah, somehow assumed the constructor got it by ref. Since it didn't, I think it's fine like this

Reply:
Yeah, I think that was actually the idea there originally - and if the constructor get's it by const ref, we can also store it by const ref, because indeed in that case initializing with a temporary in the constructor should still be fine and whenever something is actually passed down (initially to ModelChecker from what I saw), then that argument outlives this class, so it's fine to keep it a reference...

Reply:
Yea that's what I assumed before